### PR TITLE
Allow creating tenant resources via Helm chart

### PR DESCRIPTION
**Pull request by misberner**
## Description

Add a mechanism to allow creating tenant resources through a Helm chart, avoiding the need to specify these as Go structs. The chart is currently empty, I will add resources to it in a follow-up PR, and at a later point we can also add existing extra resources like routes or even the central CR to it.

The approach in this PR uses the chart only for rendering, and then creates the resources individually. For reconciliation, it merely checks if a resource of the given kind and name exists, but does not attempt to actually reconcile any diverging spec. This matches what is currently done with other resources, such as routes.

Eventually, it would be nice to use a `helm upgrade` for reconciliation, or at least have some approach that mimics what Helm does, i.e., keep a manifest of all resources rather than relying on rendering output. The reason this was not done here is the fact that the Helm action client depends on Kubernetes's `client-go`, and no the controller-runtime client. These seem to have hopelessly diverged, and it isn't possible to use the existing unit tests with the controller-runtime `fake` client when the Helm action client is used.

## Checklist (Definition of Done)
- [x] ~Unit and integration tests added~
- [x] ~Documentation added if necessary~
- [ ] CI and all relevant tests are passing
- [x] ~Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.~

## Test manual

- Unit test

### CODE REVIEW

File: fleetshard/pkg/central/reconciler/reconciler.go [hunk 1]
```diff
@@ -395,6 +414,76 @@ func (r *CentralReconciler) ensureCentralCRDeleted(ctx context.Context, central
 	return false, nil
 }
 
+func (r *CentralReconciler) ensureChartResourcesExist(ctx context.Context, remoteCentral private.ManagedCentral) error {
```
Comment (SimonBaeumer) [inline fleetshard/pkg/central/reconciler/reconciler.go:417]: Can you implement a small test for this function?

File: fleetshard/pkg/central/reconciler/reconciler.go [hunk 2]
```diff
@@ -395,6 +414,76 @@ func (r *CentralReconciler) ensureCentralCRDeleted(ctx context.Context, central
...
+		}
+		err = r.client.Create(ctx, obj)
+		if err != nil && !apiErrors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create object %s/%s of type %v: %w", key.Namespace, key.Name, obj.GroupVersionKind(), err)
+		}
+	}
+	return nil
+}
+
+func (r *CentralReconciler) ensureChartResourcesDeleted(ctx context.Context, remoteCentral private.ManagedCentral) (bool, error) {
```
Comment (SimonBaeumer) [inline fleetshard/pkg/central/reconciler/reconciler.go:449]: Can you implement a small test for this function?

**Review by SimonBaeumer — CHANGES_REQUESTED** [review]
The change looks great, I like using the Helm approach to render the resources.

**Comment by misberner** [comment]
@SimonBaeumer test added

**Comment by misberner** [comment]
@SimonBaeumer @johannes94 could you please take another look?

**Review by SimonBaeumer — APPROVED** [review]

**Pull request merged**